sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusException.java

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.amqp.exception.AmqpErrorCondition;
+import com.azure.core.amqp.exception.AmqpException;
+import com.azure.core.exception.AzureException;
+
+/**
+ * Defines {@link ServiceBusException} which has additional information about the operation that caused the
+ * error.
+ * @see ServiceBusErrorSource
+ */
+public final class ServiceBusException extends AzureException {
+    private final transient ServiceBusErrorSource errorSource;
+    private final transient ServiceBusFailureReason reason;
+    private final boolean isTransient;
+
+    /**
+     * @param throwable   for the error happened.
+     * @param errorSource indicating which api caused the error.
+     */
+    public ServiceBusException(Throwable throwable, ServiceBusErrorSource errorSource) {
+        super(throwable.getMessage(), throwable);
+        this.errorSource = errorSource;
+
+        if (throwable instanceof AmqpException) {
+            AmqpException amqpException = (AmqpException) throwable;
+            reason = getServiceBusFailureReasonFromException(amqpException);
+            isTransient = amqpException.isTransient();
+        } else {
+            reason = ServiceBusFailureReason.GENERAL_ERROR;
+            isTransient = false;
+        }
+    }
+
+    /**
+     * Gets the {@link ServiceBusErrorSource} in case of any errors.
+     * @return the {@link ServiceBusErrorSource}
+     */
+    public ServiceBusErrorSource getErrorSource() {
```

Comment:
Interesting that this is on the exception itself. Is it the same in JS?

Reply:
Ah, its not in JS. I need to move this to the ProcessErrorArgs equivalent in Java.

Reply:
An interesting thing here is that the ServiceBusProcessor in Java takes a throwable, not a context object at the moment.

Reply:
I would expect the Error Source to be available only on the context object passed to the `processError()` in the processor client builder.

Reply:
Now that the error context includes the error source, this method can be removed from this class.

Reply:
I've made this field internal but we do use it when constructing the ErrorContext. So we won't expose it from the exception but we will use it internally to fill out the context (when appropriate).